docs/src/Provider.doc.js

```diff
@@ -19,65 +19,179 @@ card(
+        ],
+        href: 'OnNavigation',
       },
     ]}
   />,
 );
 
 card(
-  <Example
-    description="Specify a light or dark color scheme for components"
-    name="Color scheme"
-    id="colorScheme"
-    defaultCode={`
+  <MainSection name="Variants">
+    <MainSection.Subsection
+      title="Color scheme"
+      description="Specify a light or dark color scheme for components"
+    >
+      <MainSection.Card
+        cardSize="lg"
+        defaultCode={`
 function Example(props) {
   const [scheme, setScheme] = React.useState('light')
   const schemeOptions = [
     {
-      value: "light",
-      label: "Light"
+      value: 'light',
+      label: 'Light'
     },
     {
-      value: "dark",
-      label: "Dark"
+      value: 'dark',
+      label: 'Dark'
     },
     {
-      value: "userPreference",
-      label: "User Preference"
+      value: 'userPreference',
+      label: 'User Preference'
     }
   ];
   return (
-    <Provider colorScheme={scheme} id="docsExample">
-      <Box color="white" padding={2}>
+    <Provider colorScheme={scheme} id='docsExample'>
+      <Box color='white' padding={2}>
         <SelectList
-          id="scheme"
-          name="scheme"
+          id='scheme'
+          name='scheme'
           onChange={({ value }) => setScheme(value)}
           options={schemeOptions}
-          placeholder="Select color scheme"
-          label="Color scheme"
+          placeholder='Select color scheme'
+          label='Color scheme'
           value={scheme}
         />
         <Box padding={2}>
           <Text>Some content</Text>
         </Box>
-        <Button text="Example button" inline /> <Button color="red" text="Red Button" inline />
+        <Button text='Example button' inline /> <Button color='red' text='Red Button' inline />
       </Box>
     </Provider>
   );
 }`}
-  />,
+      />
+    </MainSection.Subsection>
+    <MainSection.Subsection
+      title="OnNavigation"
+      description="This example illustrates the implemention of `onNavigation` context so that the link functionality of Gestalt components can be controlled externally. This example had 4 rellevant parts: a Provider, an `onNavigation` high-order function, consumer components (Link, Button, IconButton, TapArea), and `onNavigationOptions` props. The top Provider will pass to the consumer components the `onNavigation` function which will make accessible a final function that will be called during the `onClick` event handler. The `onNavigation` function can contain complex logic including React hooks to perform side effects. In this case, we are using `onNavigation` to disable the link, show an alert message, and open a different url in a new window. Finally, the `onNavigationOptions` props provides a flexible API. In this case, `navigationMode` allows to disable/enable the `onNavigation` logic. Other uses could be accessing `event.stopPropagation`."
```

Comment:
Consider breaking this up into two chunks for readability. Also perhaps copy-paste to a word processor for spellcheck and grammar check, then paste back here

Reply:
DONE